nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/deck_guns_construction.dm

```diff
@@ -0,0 +1,456 @@
+	icon_state = "platform"
+	anchored = TRUE
+	density = TRUE
+	pixel_x = -43
+	pixel_y = -64
+	bound_width = 96
+	bound_height = 128
+	bound_x = -32
+	bound_y = -64
+
+	var/state = BS_MOUNT_UNBOLTED
+	var/max_barrels = 1
+	var/igniters_added = 0
+	var/barrels_added = 0
+	
+	var/num_sheets_frame = 20
+	var/num_sheets_casing = 2
+	var/num_cables = 4
+	var/output_path = /obj/machinery/ship_weapon/deck_turret
+
+/obj/structure/ship_weapon/artillery_frame/Initialize()
+	..()
+	if(!contents)
+		contents = list()
+
+/obj/structure/ship_weapon/artillery_frame/examine(mob/user)
+	. = ..()
+	var/sare = (max_barrels != 1) ? "s are" : " is"
+	var/thit = (max_barrels != 1) ? "they" : "it"
+	//shorthands for if barrels are plural or not
+	switch(state)
+		if(BS_MOUNT_UNBOLTED)
+			. += "It is <i>unbolted</i> from the floor and could be <b>cut apart</b>."
+		if(BS_MOUNT_BOLTED)
+			. += "It is <b>bolted</b> down and can be <i>welded</i> to the floor."
+		if(BS_MOUNT_WELDED)
+			. += "It is <b>welded</b> to the floor and the <i>firing electronics</i> are missing."
+		if(BS_ELECTRONICS_LOOSE)
+			. += "The firing electronics sit loose in the frame. They could be <b>pried out</b> or <i>screwed</i> into place."
+		if(BS_ELECTRONICS_SECURE)
+			. += "The firing electronics are <b>screwed</b> into place. There is a slot for a <i>multibarrel upgrade</i>, or an <i>igniter</i>."
+		if(BS_UPGRADE_LOOSE)
+			. += "The multibarrel upgrade is loose in the frame. The <i>screws</i> are loose, and it can be <b>pried out</b> of the frame."
+		if(BS_UPGRADE_SECURED)
+			var/igniters_left = max_barrels - igniters_added
+			. += "The multibarrel upgrade is <b>screwed</b> into place. There [(igniters_left != 1) ? "are [igniters_left] slots for <i>igniters</i>." : "is 1 slot for an <i>igniter</i>."]"
+		if(BS_IGNITERS_PLACED)
+			. += "The igniter[sare] unsecured, and [thit] can either be <b>removed</b> by hand, or <i>screwed</i> into place."
+		if(BS_IGNITERS_SECURED)
+			. += "The igniter[sare] <b>screwed</b> in place, but lack <i>wiring</i>."
+		if(BS_WIRED)
+			. += "The igniter[sare] <b>wired</b> to the electronics and the wiring can be <i>soldered<i/>."
+		if(BS_WIRES_SOLDERED)
+			. += "The wiring has been <b>soldered</b>, and can be covered with a <i>plasteel casing</i>."
+		if(BS_CASING_ADDED)
+			var/barrels_left = max_barrels - barrels_added
+			. += "The plasteel casing can be <b>cut away</b>, and there [(barrels_left != 1) ? "are [barrels_left] slots for <i>barrels</i>." : "is 1 slot for a <i>barrel</i>."]"
+		if(BS_BARRELS_PLACED)
+			. += "The barrel[sare] loose in the frame, and [thit] can either be <i>welded<i> to the frame or <b>pried out</b>."
+		if(BS_BARRELS_SECURED)
+			. += "The barrel[sare] <b>welded</b> to the frame, and the assembly can be finished with a <i>loading tray</i>."
+
+/obj/structure/ship_weapon/artillery_frame/proc/set_final_state()
+	state = BS_BARRELS_SECURED
+
+/obj/structure/ship_weapon/artillery_frame/attackby(obj/item/W, mob/user, params)
```

Comment:
Most expensive attackby proc I've ever seen 😆 

if you use a switch statement, you can cover most of the state checks, then check types from there